File: 118-pascals-triangle/pascals-triangle.py
class Solution(object):
    def generate(self, numRows):
        """
        :type numRows: int
        :rtype: List[List[int]]
        """
        final=[]
        for i in range(numRows):
            l1 = [1] * (i + 1)  
            final.append(l1)
        for i in range(2,numRows):
            for j in range(1,i):
                final[i][j]=final[i-1][j-1]+final[i-1][j]
        return final        
                    





        
        

File: 118-pascals-triangle/pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> finalList = new ArrayList<>();

        // Create rows filled with 1's
        for (int i = 0; i < numRows; i++) {
            List<Integer> row = new ArrayList<>();

            for (int j = 0; j <= i; j++) {
                row.add(1);
            }

            finalList.add(row);
        }

        // Fill the middle elements
        for (int i = 2; i < numRows; i++) {
            for (int j = 1; j < i; j++) {
                finalList.get(i).set(
                    j,
                    finalList.get(i - 1).get(j - 1) + finalList.get(i - 1).get(j)
                );
            }
        }

        return finalList;
    }
}